### ISSUE TITLE
Question about input loss function ?

### ISSUE THREAD

**Issue by ThorPham** [KevinMusgrave/pytorch-metric-learning#509]
Can I use loss metric learning like torch.nn.CosineEmbeddingLoss . I have fixed pairs of positive and negative with label [-1,1]
Input1: (N, D) 
Input2: (N, D)
Target: (N) with label 1 or -1
Thank you.

**Comment by KevinMusgrave** [comment]
So if target[i] = -1, then input1[i] and input2[i] form a negative pair, and otherwise it's a positive pair. If I understand correctly, then this should work:

```python
loss_fn = ContrastiveLoss()

p = torch.where(target==1)[0]
n = torch.where(target==-1)[0]
loss = loss_fn(input1, indices_tuple=(p, p, n, n), ref_emb=input2)

```

**Comment by KevinMusgrave** [comment]
Yes the above code works with `input1` and `input2`. As you can see, `input2` is passed in as the `ref_emb`.

**Comment by ThorPham** [comment]
@KevinMusgrave So thank you for your answer .
Can you explain indices_tuple ?
I my model i want to compute ContrastiveLoss between image and text. But also want compute ContrastiveLoss only image.
i have image with class  like this
image-1  0
image-2 0
image-3 1
image-4 2
image-5 5
How can i handle indices_tuple in this case. Thank you

**Comment by KevinMusgrave** [comment]
If you have labels, you don't need to provide `indices_tuple`. You can do `loss_fn(emb, labels)`

**Comment by ThorPham** [comment]
@KevinMusgrave it automatically generates positive and negative pairs ?

**Comment by KevinMusgrave** [comment]
Yes

**Comment by ThorPham** [comment]
I appreciate you so much
